### ISSUE TITLE
Opening file directly with Vim in command-line causes key presses to not work in PowerShell

### ISSUE THREAD

**Issue by FilBot3** [vim/vim-win32-installer#193]
# The Issue

I forget what version this started with, but it's still happening here too. With the latest at the time of this post

* https://github.com/vim/vim-win32-installer/releases/tag/v8.2.1632

When I am using both Windows Terminal with PowerShell Core or Cmder with PowerShell core and opening a file directly with vim, I cannot enter any commands into Vim. Then I have to restart the terminal.

However, I can start vim, then use the `:e` or `:Explore` commands to open files after that. 

I can use plain ol' `cmd.exe` though and open a file directly. 

# Steps to reproduce

## Install gvim

I download gvim from the url above, using the `gvim_8.2.1632_x64_signed.exe` and install it by double clicking on it, doing a full install and accepting the rest.

## Running vim

I open Windows Terminal that was installed from the Microsoft Store or Cmder downloaded from https://cmder.net and then run

```bash
vim
```

I can then open files and explore around.

However, if I use

```bash
vim .\file.json
```

on the second time I open that file, I cannot use the keybindings to navigate nor the arrow keys. The same thing happens in Cmder. 

### Cmd.exe

Using vim on `cmd.exe` seemed to work just fine consistently.

### Plain `pwsh` and PowerShell

The same issues happened here as well.

## using `--clean`

Using the `--clean` option did seem to make it work, and consistently. So, obviously there's a problem with the plugins I'm using on Windows. I was able to get `vim` to work once without the `clean` option and then it wouldn't let me navigate with the keybindings.

# Version Information

## Windows

```powershell
❯ systeminfo.exe | Select-String -Pattern:"^OS.*"

OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19041 N/A Build 19041
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
```

## Vim version

```bash
❯ vim --version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Sep  6 2020 22:02:54)
MS-Windows 64-bit console version
Included patches: 1-1632
Compiled by appveyor@APPVYR-WIN
```

## PowerShell Core Version

```powershell
❯ $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

## PowerShell Framework Version

```powershell
❯ $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by chrisbra** [comment]
please provide complete steps to reproduce. I suppose this only happens from the powershell prompt?

**Comment by k-takata** [comment]
I couldn't reproduce this.
Does it occur when you execute vim with `--clean` option?
Does it occur only when you use PowerShell Core on Windows Terminal (or Cmder)?
How about executing vim on PowerShell Core *without* using Windows Terminal? Or how about executing vim on cmd.exe on Windows Terminal?

What are the versions of the following software?
* Windows
* Windows Terminal
* PowerShell Core

**Comment by FilBot3** [comment]
I'll update the initial issue section.

**Comment by heaths** [comment]
I'm seeing this sporadically with 8.2.1724 (or at least, I think that's the version - says "Included patches: 1-1724". I don't see that behavior with 8.2.1034 on another machine and will be downgrading to that version for now. My list of plugins on both machines is exactly the same (my ~/vimrc is a git repo with submodules for bundles - one of the first things I clone on any new machine).

**Comment by chrisbra** [comment]
Can you provide a reproduction steps?

**Comment by heaths** [comment]
I can't give you anymore than I did: with that version, opening .cs files with the omnisharp-vim bundle enabled *sometimes* hangs waiting for input. Which file doesn't seem to matter. I can kill vim and reopen the same file and it works, or I can open a file in a complete different repo and get the same behavior (works or it doesn't, even on the same file).

Would a trace or a dump help at all? I can install a copy of the version I had and try to get you one that way if it would help.

**Comment by heaths** [comment]
For omnisharp-vim, see https://github.com/OmniSharp/omnisharp-vim#:~:text=OmniSharp.%20OmniSharp-vim%20is%20a%20plugin%20for%20Vim%20to,including%20Visual%20Studio%20Code%2C%20Emacs%2C%20Atom%20and%20others.. That said, I'm not even sure that's where the problem has been. The hang can happen either before rending the buffer or after. I just don't open many more files than small, quick fixes of .cs files typically (VSCode has replaced vim for most of my use, both of which are way faster than VS for small code changes).

**Comment by vimday** [comment]
Me Too

**Comment by k-takata** [comment]
I still cannot reproduce this.

Does it occur with vim.exe on cmd.exe without Windows Terminal?
Does it occur with vim.exe on cmd.exe on Windows Terminal?
Does it occur with vim.exe on PowerShell without Windows Terminal?
Does it occur with vim.exe on PowerShell on Windows Terminal?

Can someone bisect this?

**Comment by k-takata** [comment]
If you are using 8.2.1790 or later, it has a problem which is different from the original report.
See https://github.com/vim/vim/issues/7083.

**Comment by heaths** [comment]
@k-takata that could be the problem I'm experiencing. I do have some python in my vimrc. I will try to find some time to install a newer version, remove that, and see if it ever repros.

**Comment by k-takata** [comment]
The problem in 8.2.1790 has been fixed in 8.2.1806. (But this might be different from the original issue.)

**Comment by k-takata** [comment]
The original issue should be fixed in 8.2.1830 (https://github.com/vim/vim/commit/c6ed254d9fda0ff54cdedce5597ff3e0d0218d18).
